frontend/app/component/form/candidates_votes_form/CandidatesVotesForm.tsx

```diff
@@ -171,10 +171,9 @@ export function CandidatesVotesForm({ group }: CandidatesVotesFormProps) {
 
 function candidateNumberFromElement(el: HTMLInputElement) {
   const id = el.id;
-  const bits = id.split("[");
-  const numberString = bits[bits.length - 1]?.split("]")[0];
-  if (numberString) {
-    return parseInt(numberString) + 1;
+  const regexMatchArray = id.match(/[(\d*)]/g);
+  if (regexMatchArray) {
+    return parseInt(regexMatchArray[0]) + 1;
```

Comment:
Two thoughts, but they apply to the previous implementation as well, so probably better left for another PR.
1. If we change the function's parameter from an HTMLInputElement to the id, we can write some tests for it.
2. If there is no match, we return `0`. And then what happens? What if there are multiple no matches?

Reply:
Good call on the testing, because I'm terrible at regex. Added tests and fixed a bug in c9bc125fe7102ff7b31d62d00ec7925003ea1b4e

Reply:
And to answer your question 2. Candidate number 0 does not exist, so then there will be an `IncorrectCandidatesList` validation error from the backend. This should never happen of course, just like the OutOfRange validation error, but they are implemented.

Reply:
We're missing a frontend test for the `IncorrectCandidatesList`. I'll create a PR to add it.